### PR TITLE
adds jss theme provider

### PR DESCRIPTION
**Pull request by 3axap4eHko**

### CODE REVIEW

File: packages/gatsby-plugin-jss/src/gatsby-browser.js [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
 exports.onInitialClientRender = () => {
   // eslint-disable-next-line no-undef
   const ssStyles = window.document.getElementById(`server-side-jss`)
   ssStyles && ssStyles.parentNode.removeChild(ssStyles)
 }
+
+exports.wrapRootComponent = ({ Root }, options) => () => {
+  const theme = options.theme || {}
+  return (
+    <ThemeProvider theme={theme}>
```
[the remark this answers is not in the document]
Comment (3axap4eHko) [reply packages/gatsby-plugin-jss/src/gatsby-browser.js]: I didn't find them except one more HOC in an app

File: packages/gatsby-plugin-jss/src/gatsby-browser.js [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
 // remove the JSS style tag generated on the server to avoid conflicts with the one added on the client
 exports.onInitialClientRender = () => {
   // eslint-disable-next-line no-undef
   const ssStyles = window.document.getElementById(`server-side-jss`)
   ssStyles && ssStyles.parentNode.removeChild(ssStyles)
 }
+
+exports.wrapRootComponent = ({ Root }, options) => () => {
+  const theme = options.theme || {}
```
[the remark this answers is not in the document]
Comment (3axap4eHko) [reply packages/gatsby-plugin-jss/src/gatsby-browser.js]: Done

**Comment by m-allanson** [comment]
Oh, this looks good but I have one more request: 

Gatsby recently added a requirement that contributors sign-off commits. Could you amend your commits to include a DCO signoff? 

To signoff your most recent commit use `git commit --amend --signoff`, for a whole branch see [this Stack Overflow answer](https://stackoverflow.com/questions/13043357/git-sign-off-previous-commits#27387391).

For more info check out the new [Developer Certificate of Origin docs](https://www.gatsbyjs.org/docs/how-to-contribute/#developer-certificate-of-origin) or comment here if you've any questions.

**Comment by 3axap4eHko** [comment]
@m-allanson I'm trying to signoff entire branch but getting this error
```
fatal: cannot run gpg: No such file or directory
error: gpg failed to sign the data
could not write rewritten commit
```
So could we skip it for now?

**Comment by m-allanson** [comment]
Oh hmmm. Maybe squash everything to one commit and do `git commit --amend --signoff`?

**Comment by KyleAMathews** [comment]
Just skipped over DCO :-\ we'll probably switch to using a CLA instead next week.

**Pull request merged**